### DISCUSSION TITLE
MacOS Notarization

### DISCUSSION THREAD

**Discussion by unfamiliartunes** [pierreguillot/Camomile#267] [Q&A]
Hi!
Some of my MacOS beta testers are having problems using my patch/plugin because of the "unsigned developer" issues that Big Sur creates. Is there a fix for that, is it in development, or is it simply impossible to bypass? Thank you!

**Comment by pierreguillot** [comment]
The plugin is not code-signed. I don't have an Apple developer account for that and even if I have I'm not sure it will work because if the bundle is modified by introducing or modifying new files (such as integrating your Pd patch in the bundle), the code-signing is broken and might lead to a crash. I think the solution is to recompile the plugin with your patch (and all the necessary files) in the bundle and to code-sign it yourself... here is the doc: https://developer.apple.com/library/archive/documentation/Security/Conceptual/CodeSigningGuide/Introduction/Introduction.html#//apple_ref/doc/uid/TP40005929

**Comment by Reinissance** [comment]
Not necessary: Once You tried opening the plugin and it fails you have to go to the settings app, to Security, and when unlocked you should see a hint that the plugin was blocked with the option to „allow though“ or something. After doin so, once you try to use the plugin again you are promted again whether you still want to open it and thereafter the system remembers that you allowed the plugin to run on your machine. Thus I got it running on macOS 11.11.

**Reply by pierreguillot** [reply]
I changed the name of the discussion to be more explicit. 
The apple notarization becomes really annoying for this kind of project. If someone has an idea to avoid the back and forward with the security panel to accept the plugins, let me know. even if I prefer not to, I can also buy an Apple developer account but I prefer t be sure that it will work in this situation.

**Comment by pierreguillot** [comment]
I think I might have a solution, I can add a line in the Camomile script to remove the quarantine when building the plugin.

**Reply by pierreguillot** [reply]
As explained in this issue (https://github.com/pierreguillot/Camomile/issues/280), I found a workaround. If you can download and test the [binaries](https://github.com/pierreguillot/Camomile/actions/runs/1408034867), it would be useful. You have to run the camomile script to generate the plugins (using `sudo` might be necessary, for example: `sudo ./camomile`).
